### DISCUSSION TITLE
Bug on BaseUrl.php that prevent parameters being tracked in the URL?

### DISCUSSION THREAD

**Discussion by damib** [livewire/livewire#9078] [1. Help]
### Livewire version

3.5.17

### Laravel version

11.35.0

### Which PHP version are you using?

PHP 8.4

### What is the problem?

the #[Url] does not work properly: it allows to initialize the parameters from the URL but doesn't in the other direction so that the URL never tracks the parameter changes. Looking in the code I've found that commenting the following line in the file BaseUrl.php solves the problem:

```php

public function dehydrate($context)
    {
        #if (! $context->mounting) return;

        $this->pushQueryStringEffect($context);
    }
```

I'm sure this is not the correct solution because this line of code has been there for long time, but this may help someone to understand what's going on...

### Code snippets

The problem is there from a fresh laravel installation. (I've tried twice)

### Wirebox URL

no wirebox

### How do you expect it to work?

no wirebox

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my help request will be removed if I haven't met the criteria above.

**Comment by damib** [comment]
The problem goes away if instead of #[Url] I use the:

```php
    protected function queryString()
    {
        return [
            'search' => [],
        ];
    }

```
In the component class.

**Comment by damib** [comment]
UPDATE: Every problem disappear if I prevent the loading of the javascript assets from Vite:

```php
        {{-- @vite(['resources/css/app.css', 'resources/js/app.js']) --}}
        @vite(['resources/css/app.css'])

```
Is there something in the default configuration of laravel assets that interferes with livewire machinery?

**Comment by damib** [comment]
From the Livewire docs:

`
Livewire ships with Alpine out of the box so there is no need to install it into your project separately.
`
From the default laravel /resurces/js/app.js:

```
import './bootstrap';

import Alpine from 'alpinejs';

window.Alpine = Alpine;

Alpine.start();
``` 

Is Alpine included twice in the latest laravel 11 Livewire versions? The problem goes away as soon as I comment the following lines:

```js
import './bootstrap';

// import Alpine from 'alpinejs';

// window.Alpine = Alpine;

// Alpine.start();
```
And after that Alpine is still working!!

**Reply by joshhanley** [reply]
@damib Laravel doesn't ship with Livewire or Alpine by default. So how did you install it? If you used Breeze and selected the Blade view option, it will install Alpine in your `app.js` as it's expected that you're not using Livewire. If you decide to pull Livewire in later, you need to remove Alpine from your `app.js` as you've discovered otherwise it won't function correctly.
